curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -166,12 +166,26 @@ public static String configToConnectionString(QuorumVerifier data) throws Except
         StringBuilder sb = new StringBuilder();
         for ( QuorumPeer.QuorumServer server : data.getAllMembers().values() )
         {
+            if ( server.clientAddr == null )
+            {
+                // Invalid client address configuration in zoo.cfg
+                continue;
```

Comment:
I see what you mean - if the client_port_address is not present ZK will not initialize server.clientAddress, which is where you extract the client port. This is not a problem for ZK itself because it has the knowledge of clientPort of a quorum peer which is not publicly accessible.

I think QuorumSever can provide APIs so you guys can use the APIs to get the client / server address / port rather than relying on the clientAddr etc fields. Before that happens it's fine to skip as there is no way to extract the client port (and I think that only happens when clientAddress is not explicitly present in the config string - cases like server.1= 125.23.63.23:1234:1235;1236)